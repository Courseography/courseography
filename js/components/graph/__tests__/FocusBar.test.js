import { fireEvent } from "@testing-library/react"
import TestFocusBar from "./TestFocusBar"
import TestContainer from "./TestContainer"

describe("FocusBar", () => {
  it("should match shallow snapshot", async () => {
    const focusBar = await TestFocusBar.build()
    expect(focusBar).toMatchSnapshot()
  })

  it("FocusBar should be disabled for non-computer science graphs", async () => {
    const container = await TestContainer.build()
    expect(container.getByRole("button", { name: /FOCUSES ткв/i })).toBeInTheDocument
    fireEvent.click(container.getByTestId("test-graph-1"))
    expect(container.queryByRole("button", { name: /FOCUSES ткв/i })).toBeNull
  })

  it("Pressing the FocusBar button should open it", async () => {
    const focusBar = await TestFocusBar.build()
    expect(focusBar.queryByRole("button", { name: /Scientific Computing/i })).toBeNull
    fireEvent.click(focusBar.getByRole("button", { name: /FOCUSES ткв/i }))
    expect(focusBar.getByRole("button", { name: /Scientific Computing/i }))
      .toBeInTheDocument
  })
})
