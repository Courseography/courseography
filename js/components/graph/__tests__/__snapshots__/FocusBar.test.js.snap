// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FocusBar should match shallow snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    

    <nav>
      
    
      <ul>
        
        
        <li
          id="nav-graph"
        >
          
            
          <a
            href="/graph"
          >
            Graph
          </a>
          
        
        </li>
        
    
      </ul>
      

    </nav>
    

    <div
      class="react-graph"
      id="react-graph"
    />
    

    <div
      id="fcecount"
    />
    <div>
      <div
        class="focus-menu-bar"
      >
        <button
          class="focus-menu-toggle"
        >
          Focuses ткв
        </button>
        <div
          class="focuses-list"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="focus-menu-bar"
    >
      <button
        class="focus-menu-toggle"
      >
        Focuses ткв
      </button>
      <div
        class="focuses-list"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
